### PR TITLE
[FEATURE] Make map tips delay configurable

### PR DESCRIPTION
**Pull request by jdugge**
## Description
This PR makes the delay before showing a MapTip configurable. At the moment, the delay is fixed at 850 milliseconds. With this PR, the delay can be configured using `qgis/mapTipsTimerInterval` in the Advanced Settings Editor.

## Checklist

- [x] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [x] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by m-kuhn** [comment]
Good move.
What value did you change it to? Would it make sense to adjust the default instead?
I wonder if it would make sense to make this more accessible. The way it's done now requires a big knowledge of the internals. Might be worth at least writing the setting where it doesn't exist to have it listed in the editor.

**Comment by jdugge** [comment]
I actually set it to `0` because I needed to quickly compare different points, but that's obviously not a sensible default value.

I'm happy to adjust the pull request to make the setting more accessible. Would the right place for this be the "Options / Map Tools" mask, creating a new field set "Map set"?

**Comment by m-kuhn** [comment]
That sounds good.
Maybe put it into "Map Canvas / Legend" in a new group "Map Tips"?

**Comment by jdugge** [comment]
I've added the setting to the menu.
However, at the moment the setting only takes effect after restarting QGIS. I assume this happens because  `QgisApp::createMapTips()` is only run at startup. What's the correct place for re-initializing the map tips? Is there a function that's called after settings have changed?

**Comment by m-kuhn** [comment]
You will need setter/getter in QgisApp that take care of updating the setting and adjusting the internals. If you want to get fancy you can also add it to QgisInterface so plugins can also change this without requiring a restart. If you want to get extra-fancy you can emit a signal whenever the setting is changed.

**Comment by jdugge** [comment]
Thank you! I've taken the first step, creating a setter `QgisApp::setMapTipsDelay` in `qgisapp.cpp` that gets called from `QgsOptions::saveOptions()`. Is this the correct approach?

**Comment by jdugge** [comment]
@m-kuhn Can this be merged in it's current form, or does it still need changes?

**Comment by jdugge** [comment]
@m-kuhn Sorry that I keep pushing this, but could you indicate if I need to make changes to this to get it accepted?

File: src/app/qgisapp.h [hunk 1]
```diff
@@ -941,10 +941,8 @@ class APP_EXPORT QgisApp : public QMainWindow, private Ui::MainWindow
      */
     void reloadConnections();
 
-    /**
-     * Shows the layout manager dialog.
-     * \since QGIS 3.0
-     */
+    void setMapTipsDelay( int timerInterval );
+
     void showLayoutManager();
```
Comment (m-kuhn) [inline src/app/qgisapp.h]: Oops, a docstring got lost here

File: src/app/qgisapp.h [hunk 2]
```diff
@@ -941,10 +941,8 @@ class APP_EXPORT QgisApp : public QMainWindow, private Ui::MainWindow
      */
     void reloadConnections();
 
-    /**
-     * Shows the layout manager dialog.
-     * \since QGIS 3.0
-     */
+    void setMapTipsDelay( int timerInterval );
```
Comment (m-kuhn) [inline src/app/qgisapp.h]: Missing docstring

**Comment by m-kuhn** [comment]
It looks quite good to me. I guess travis is mainly failing for api doc reasons (see my comments) and then it's good to go for 3.2.

sorry for the delay, I am out of office since Jan and until mid-March.

**Comment by jdugge** [comment]
Thanks for the feedback! I've fixed the missing docstring, and the tests all pass now.

Sorry for the mess with the multiple commits, I'm not quite sure what the proper way to update a pull request (`rebase` then `push --force`?)

**Comment by jdugge** [comment]
@m-kuhn Will this be merged right away, or just before the release of 3.2?

**Pull request merged**